GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -428,45 +428,29 @@ public virtual void GetFileTypeAndMode(string filePath, out FileType fileType, o
             }
         }
 
-        public virtual PathSparseState GetPathSparseState(string virtualPath)
+        public virtual PathSparseState GetFolderPathSparseState(string virtualPath)
         {
-            this.GetChildNameAndParentKey(virtualPath, out string fileName, out string parentKey);
-            FolderEntryData data = this.GetProjectedFolderEntryData(
-                blobSizesConnection: null,
-                childName: fileName,
-                parentKey: parentKey);
-
-            if (data == null)
-            {
-                return PathSparseState.NotFound;
-            }
-            else if (data.IsIncluded)
-            {
-                return PathSparseState.Included;
-            }
-            else
+            if (this.TryGetFolderDataFromTreeUsingPath(virtualPath, out FolderData folderData))
             {
-                return PathSparseState.Excluded;
+                return folderData.IsIncluded ? PathSparseState.Included : PathSparseState.Excluded;
             }
+
+            return PathSparseState.NotFound;
         }
 
         public bool TryAddSparseFolder(string virtualPath)
         {
             try
             {
-                this.GetChildNameAndParentKey(virtualPath, out string fileName, out string parentKey);
-                FolderEntryData data = this.GetProjectedFolderEntryData(
-                    blobSizesConnection: null,
-                    childName: fileName,
-                    parentKey: parentKey);
-
-                if (data != null && !data.IsIncluded)
+                if (this.TryGetFolderDataFromTreeUsingPath(virtualPath, out FolderData folderData) &&

```

Comment:
It looks like with this change we'll be skipping the lookup in the cache that would happen in `GetProjectedFolderEntryData`.  Do we know at this point that this folder won't be in the cache?

(I am guessing that because we're in `TryAddSparseFolder` we'd know the folder is not in the cache?)

Adding a comment about out expectations of the cache would be helpful.

Reply:
This is for the create new folder scenario and we checked the sparse state and got back excluded in order to get to this code and excluded folders do not get added to the cache.  I will add some comments.